frontend/src/product/components/NavBar.js

```diff
@@ -0,0 +1,102 @@
+import React, { useEffect, useState } from "react";
+import { GiMeat } from "react-icons/gi";
+import { GiFruitBowl } from "react-icons/gi";
+import { FaLeaf } from "react-icons/fa";
+import { GiCook } from "react-icons/gi";
+import { GiMilkCarton } from "react-icons/gi";
+import { GiMuscleFat } from "react-icons/gi";
+import { FaShoppingCart } from "react-icons/fa";
+import ReactTooltip from 'react-tooltip';
+import ProductItem from './ProductItem';
+import {  Label } from "semantic-ui-react";
+import Cart from '../pages/Cart'
+let nextIndex = 3;
+let prevIndex = 0;
+const NavBar = ({ products }) => {
+    const [product, setProduct] = useState(products);
+    const [cartItems, setCartItems] = useState([]);
+    const [productNum, setProductNum] = useState(0);
+    const cartItem = (item) => {
+        setCartItems([...cartItems, item])
+    }
+    const addToCart = (num) => {
+        setProductNum(num);
+    }
+    useEffect(() => {
+        setProduct(products);
+    }, [products])
+    const next = () => {
+        setProduct(products.filter((pro, index) => index < nextIndex && index >= prevIndex && index < products.length));
```

Comment:
This part of code(between Line 29 - Line 37) looked a bit complex to me. Could you elaborate on that?